src/time_series_predictor.py

```diff
@@ -0,0 +1,154 @@
+"""
+time_series_predictor script
+"""
+import datetime
+import warnings
+
+import numpy as np
+import psutil
+import torch
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+# Show switch to cpu warning
+warnings.filterwarnings("default", category=ResourceWarning)
+
+# pylint: disable=too-many-instance-attributes
+class TimeSeriesPredictor:
```

Comment:
For a real development project, you could argue this class makes the code cleaner, but in our case it just adds more code without real change in the results.

Reply:
Indeed.